database-connector/src/main/java/life/qbic/projectmanagement/experiment/persistence/QbicSampleDataRepo.java

```diff
@@ -0,0 +1,45 @@
+package life.qbic.projectmanagement.experiment.persistence;
+
+import java.util.List;
+import life.qbic.projectmanagement.domain.project.Project;
+import life.qbic.projectmanagement.domain.project.sample.Sample;
+import life.qbic.projectmanagement.domain.project.sample.SampleCode;
+
+
+/**
+ * <b>Sample Data Storage Interface</b>
+ *
+ * <p>Provides access to the persistence layer that handles the {@link Sample} data storage.
+ *
+ * @since 1.0.0
+ */
+public interface QbicSampleDataRepo {
+
+  /**
+   * Creates a reference to one or more {@link Sample}s in the data repository to connect project data.
+   * A project must be provided, a unique experiment is created with each batch.
+   *
+   * @param project the {@link Project} for which samples should be created
+   * @param sampleBatch the batch of {@link Sample}s to be created in the data repo
+   * @since 1.0.0
+   */
+  void addBatch(Project project, List<Sample> sampleBatch);
```

Comment:
I think it is not a batch, since we dont care for it in openBIS. For this infrastructure interface, adding samples to a project is sufficient imo.